### PR TITLE
adding models and minor changes to routes

### PR DESCRIPTION
**Pull request by Jan561**

### CODE REVIEW

File: app/controllers/root_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RootController < ApplicationController
+  def index
+    redirect_to search_search_path
```
Comment (hargoniX) [inline app/controllers/root_controller.rb:3]: This is not required and can be done in config/routes.rb

File: config/credentials.yml.enc [hunk 2]
```diff
@@ -1 +1 @@
-Ge9m5jzbKw7yOcXzwoKo4TM/dNWuyELOCwyaacQj9JJ0tsyVrV4HM8byeyazm01X3FT5vFaYrjdl8IFCn2C4zcuIZGIK1fiZiyuHhW/6Hc/Z0onVN3jJMWFakW/zHyb4yh5VCUURo9XW0st7tvwZzrATYeTNIMTcVHWyWCRtQ4pRIw0meX7+rtmnUykc38oZqOrAZUQdqEZTkEVjl0sE1biXlaGmVI15aWpO9JEb+35J2m8/3YlZLi/0nF0QdiSXqWNA4Pn0wayu8gSABxrO73VrBMLKAdN2K+Wts3OGJhudyc/eY5XDrnQN/ea0+1l8fN33vrqlDvFMSoDmkazB40dMuVFU1O6rVS8lOqDsRNa8GGKKYSik0HRjQKfnsHZZSR5LITy9xYJikw3NsjBqhPYaahzRlclOfkWT--VRKRBGFlvFwMFZxo--0c/M1rH0h4UwrOWgM31Nzg==
\ No newline at end of file
+CzYqQPZHZ0XHWe1DkMwdmiKmzk2kC6D53xkX+oUUP5iMq5u+Ml2qvgdpz/5PB7enzwPNAIBxadB+FU3ZHQLicP5nXaevJeCl5IMr2XQPU+h0PG3+FozThAjipSq60mfgdMPi8jkxHvAd56DISnuYKGG66n6gC2DqSwfFbhLVP/AajWCbrTzVvPVkAyE/6noIY7UrgKPi/TTvtngfuLKZFt3BwulFvEic/s1KHutZVtBUwmz57a8208W70HFkdhoGJsZvk9ODbLZJtML7LvZJSKBUcN7jm5Z30OZMEe0jgDYQbKFzHC0pLWnF8OSdSGlr7rHo/OwVkj+UY3lTTVtebcff9vHeL3i6VsKLlgFYgorgKViET+5GpFtBsFWaI8sC1U6TzppRrGQRLvnff3m3dwnY+2zwEJbn9lok--xrqz7GpwwDLVHT0+--jGQSXkspsgJDEdGo23BFmg==
```
Comment (hargoniX) [inline config/credentials.yml.enc:1]: What does this file even do, do we want to track it with git?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
+  root 'root#index'
```
Comment (hargoniX) [inline config/routes.rb:2]: Why not just leave it be and add a new controller for this? If the makers of this DSL didn't want us to use the root directive and instead redirect they would've done so

File: db/migrate/20200506102836_create_judgments.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      t.string :additional_file_numbers
+
+      # nil if redacted
+      t.string :plaintiff
+      t.string :defendant
+
+      # Example: Kannibale von Rotenburg
+      t.string :alias
+
+      t.string :judges, null: false
```
Comment (hargoniX) [inline db/migrate/20200506102836_create_judgments.rb:17]: Judges implies a plural, if we actually want to track multiple judges here we'd have to do an n<>m relation with 2 more tables, not exactly sure wether it's reasonable.

If we want to do it without that I'd recommend just tracking the main judge.

File: db/migrate/20200506102836_create_judgments.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+      t.timestamps
+
+      t.index :file_number, unique: true
```
Comment (hargoniX) [inline db/migrate/20200506102836_create_judgments.rb:21]: Where is the date?

File: test/test_helper.rb [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 class ActiveSupport::TestCase
   # Run tests in parallel with specified workers
-  parallelize(workers: :number_of_processors)
+  parallelize(workers: 1)
```
Comment (hargoniX) [inline test/test_helper.rb:7]: Reason?

Also how is this related to the PR?

File: db/migrate/20200506102836_create_judgments.rb (hunk 4 above)
Comment (hargoniX) [reply db/migrate/20200506102836_create_judgments.rb:17]: Furthermore its questionable wether we can rely on the uniqness of names of judges here, law is a wide field it is not unlikely there have been (or are) two judges with the same name, it would be a huge hassle to track these properly as different identities.

Next question would be if we actually want to treat them like that, if it's not relevant the n<>m relation seems reasonable.

File: app/controllers/root_controller.rb (hunk 1 above)
Comment (Jan561) [reply app/controllers/root_controller.rb:3]: As of now, there are two routes pointing to the search controller, leading to redundancy and unnecessary confusion. So the question is how we wanna route to the search action, via `/` or via `/search`?

File: config/credentials.yml.enc (hunk 2 above)
Comment (Jan561) [reply config/credentials.yml.enc:1]: autogenerated when starting server in production mode. i dont know why this isn't excluded in gitignore

File: config/routes.rb (hunk 3 above)
Comment (Jan561) [reply config/routes.rb:2]: see previous discussion

File: test/test_helper.rb (hunk 6 above)
Comment (Jan561) [reply test/test_helper.rb:7]: because postgres doesn't like multiple workers for tests (at least it fixed the problem with docker tests, dunno if still required)

File: db/migrate/20200506102836_create_judgments.rb (hunk 5 above)
Comment (Jan561) [reply db/migrate/20200506102836_create_judgments.rb:23]: fixed

File: db/migrate/20200506102836_create_judgments.rb (hunk 4 above)
Comment (Jan561) [reply db/migrate/20200506102836_create_judgments.rb]: Can you take a look at the updated version?

File: db/migrate/20200506102836_create_judgments.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      t.string :additional_file_numbers, array: true, default: []
+
+      # nil if redacted
+      t.string :plaintiff
+      t.string :defendant
+
+      # Example: Kannibale von Rotenburg
+      t.string :alias
+
+      t.string :judges, array: true, null: false
```
Comment (Jan561) [inline db/migrate/20200506102836_create_judgments.rb:17]: @hargoniX Can you take a look?

File: test/test_helper.rb (hunk 6 above)
Comment (Jan561) [reply test/test_helper.rb:7]: It's still required. When using `:number_of_processors`, we get
```
Traceback (most recent call last):
	20: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:68:in `block in autorun'
	19: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:138:in `run'
	18: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `start'
	17: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `map'
	16: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `each'
	15: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `times'
	14: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `block in start'
	13: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `fork'
	12: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:95:in `block (2 levels) in start'
	11: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1138:in `method_missing'
	10: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1158:in `with_friend'
	 9: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1139:in `block in method_missing'
	 8: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1227:in `open'
	 7: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1140:in `block (2 levels) in method_missing'
	 6: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1251:in `send_message'
	 5: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:929:in `send_request'
	 4: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `send_request'
	 3: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `each'
	 2: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:611:in `block in send_request'
	 1: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:565:in `dump'
/home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:565:in `dump': no _dump_data is defined for class PG::Connection (TypeError)
	25: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:68:in `block in autorun'
	24: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:138:in `run'
	23: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `start'
	22: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `map'
	21: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `each'
	20: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `times'
	19: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `block in start'
	18: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `fork'
	17: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:95:in `block (2 levels) in start'
	16: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1138:in `method_missing'
	15: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1158:in `with_friend'
	14: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1139:in `block in method_missing'
	13: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1227:in `open'
	12: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1140:in `block (2 levels) in method_missing'
	11: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1251:in `send_message'
	10: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:929:in `send_request'
	 9: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `send_request'
	 8: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `each'
	 7: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:611:in `block in send_request'
	 6: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:564:in `dump'
	 5: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:567:in `rescue in dump'
	 4: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:650:in `make_proxy'
	 3: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:650:in `new'
	 2: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1101:in `initialize'
	 1: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1807:in `to_id'
/home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1738:in `current_server': DRb::DRbServerNotFound (DRb::DRbServerNotFound)
	25: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:68:in `block in autorun'
	24: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:138:in `run'
	23: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `start'
	22: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `map'
	21: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `each'
	20: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `times'
	19: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `block in start'
	18: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `fork'
	17: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:95:in `block (2 levels) in start'
	16: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1138:in `method_missing'
	15: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1158:in `with_friend'
	14: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1139:in `block in method_missing'
	13: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1227:in `open'
	12: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1140:in `block (2 levels) in method_missing'
	11: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1251:in `send_message'
	10: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:929:in `send_request'
	 9: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `send_request'
	 8: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:610:in `each'
	 7: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:611:in `block in send_request'
	 6: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:564:in `dump'
	 5: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:567:in `rescue in dump'
	 4: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:650:in `make_proxy'
	 3: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:650:in `new'
	 2: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1101:in `initialize'
	 1: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1807:in `to_id'
/home/jan/.rbenv/versions/2.6.5/lib/ruby/2.6.0/drb/drb.rb:1738:in `current_server': DRb::DRbServerNotFound (DRb::DRbConnError)
	11: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:68:in `block in autorun'
	10: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:138:in `run'
	 9: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `start'
	 8: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `map'
	 7: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `each'
	 6: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:74:in `times'
	 5: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `block in start'
	 4: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:75:in `fork'
	 3: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:94:in `block (2 levels) in start'
	 2: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:97:in `rescue in block (2 levels) in start'
	 1: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:97:in `each'
/home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:98:in `block (3 levels) in start': undefined method `exception=' for #<Minitest::UnexpectedError: Unexpected exception> (NoMethodError)
Did you mean?  exception
Traceback (most recent call last):
	2: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:68:in `block in autorun'
	1: from /home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/minitest-5.14.0/lib/minitest.rb:145:in `run'
/home/jan/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems/activesupport-6.0.2.2/lib/active_support/testing/parallelization.rb:118:in `shutdown': Queue not empty, but all workers have finished. This probably means that a worker crashed and 1 tests were missed. (RuntimeError)
```

Comment (Jan561) [reply test/test_helper.rb:7]: I doubt we want to open a separate pr just for this very subtle bug fix. Marking as resolved

File: app/controllers/root_controller.rb (hunk 1 above)
Comment (hargoniX) [reply app/controllers/root_controller.rb:3]: Why not both?

File: config/credentials.yml.enc (hunk 2 above)
Comment (hargoniX) [reply config/credentials.yml.enc]: Add it then?

File: db/migrate/20200506102836_create_judgments.rb (hunk 7 above)
Comment (Jan561) [reply db/migrate/20200506102836_create_judgments.rb:17]: Or do we really want to just track the main judge (not even sure if all courts have a main judge)

File: config/credentials.yml.enc [hunk 8]
```diff
@@ -1 +0,0 @@
-Ge9m5jzbKw7yOcXzwoKo4TM/dNWuyELOCwyaacQj9JJ0tsyVrV4HM8byeyazm01X3FT5vFaYrjdl8IFCn2C4zcuIZGIK1fiZiyuHhW/6Hc/Z0onVN3jJMWFakW/zHyb4yh5VCUURo9XW0st7tvwZzrATYeTNIMTcVHWyWCRtQ4pRIw0meX7+rtmnUykc38oZqOrAZUQdqEZTkEVjl0sE1biXlaGmVI15aWpO9JEb+35J2m8/3YlZLi/0nF0QdiSXqWNA4Pn0wayu8gSABxrO73VrBMLKAdN2K+Wts3OGJhudyc/eY5XDrnQN/ea0+1l8fN33vrqlDvFMSoDmkazB40dMuVFU1O6rVS8lOqDsRNa8GGKKYSik0HRjQKfnsHZZSR5LITy9xYJikw3NsjBqhPYaahzRlclOfkWT--VRKRBGFlvFwMFZxo--0c/M1rH0h4UwrOWgM31Nzg==
```
Comment (hargoniX) [inline config/credentials.yml.enc:1]: ADd this to gitignore and it should be fine.

File: db/migrate/20200506102836_create_judgments.rb (hunk 7 above)
Comment (hargoniX) [reply db/migrate/20200506102836_create_judgments.rb:17]: I guess if this works it is fine for now at least.

File: config/credentials.yml.enc (hunk 2 above)
Comment (Jan561) [reply config/credentials.yml.enc]: Done

**Comment by Jan561** [comment]
I suggest a squash merge
